src/libraries/System.Text.Json/src/System/Text/Json/Writer/RawIndentation.cs

```diff
@@ -0,0 +1,121 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System.Diagnostics;
+
+namespace System.Text.Json
+{
+    internal readonly struct RawIndentation : IEquatable<RawIndentation>
+    {
+        private readonly Memory<byte> _bytes;
+        private readonly int? _length;
+        private readonly byte? _byte;
+
+        private const byte Null = (byte)'\0';
+
+        public readonly byte Byte => _byte ?? JsonConstants.Space;
+        public readonly ReadOnlySpan<byte> Bytes => _bytes.Span;
+        public readonly int Length => _length ?? 2;
+
+        public static RawIndentation FromString(string value) => new(value);
+
+        private RawIndentation(string value)
+        {
+            if (value is null)
+            {
+                ThrowHelper.ThrowArgumentNullException(nameof(value));
+            }
+
+            byte[] indentBytes = Encoding.UTF8.GetBytes(value);
+            byte? indentByte = null;
+
+            foreach (byte b in indentBytes)
+            {
+                if (JsonConstants.IndentChars.IndexOf(b) is -1)
+                {
+                    throw new ArgumentOutOfRangeException(nameof(value), $"{nameof(value)} contains an invalid character. Allowed characters are space and horizontal tab.");
+                }
+
+                if (indentByte is not Null)
+                {
+                    byte? previous = indentByte;
+                    indentByte = b;
+                    if (previous is not null && indentByte != previous)
+                    {
+                        indentByte = Null;
+                    }
+                }
+            }
+
+            _byte = indentByte;
+            _length = indentBytes.Length;
+
+            if (indentByte is Null)
+            {
+                _bytes = indentBytes;
+            }
+        }
+
+        public readonly bool Equals(RawIndentation other) =>
+            _byte == other._byte &&
+            _length == other._length &&
+            _bytes.Span.SequenceEqual(other._bytes.Span);
+
+        public override bool Equals(object? obj) =>
+            obj is RawIndentation indentation && Equals(indentation);
+
+        public override readonly int GetHashCode()
```

Comment:
Why does it need to be equatable?

Reply:
Required by the `JsonWriterOptionsTests.JsonWriterOptions` unit test.